lib/skins.js

```diff
@@ -19,60 +18,157 @@ var concat = lazyReq('gulp-concat');
+    if (config.dev_skin) {
+      opts.skin = config.dev_skin.id;
+      debug(opts, 'found skin: ' + opts.skin);
+      var coreSkin = findCoreSkin(config, opts.skin);
+      debug(opts, 'coreSkin: ' + coreSkin);
+      if (config.skin[coreSkin]) {
+        opts.feature = config.skin[coreSkin].feature_id;
+        debug(opts, 'found feature: ' + opts.feature);
+        if (config.feature[opts.feature]) {
+          opts.version = config.feature[opts.feature].fq_version;
+          debug(opts, 'found version: ' + opts.version);
+        }
+      }
+    }
+  }
+
+  function withOptions(cb) {
+    var server = getServer();
+    var version = server.localSkinCompileVersion();
+    var opts = {
+      debugMode: gutil.env['debug'],
+      verboseMode: gutil.env['verbose'],
+      configDir: server.configDir(),
+      version: version,
+      feature: server.localSkinCompileFeature(),
+      skin: server.localSkinCompileSkin(),
+      dest: server.localSkinCompileDest(),
+      localServerDir: server.localServerDir(),
+      localServerPort: server.localServerPort(),
+      includePathsPrefix: '',
+      includePaths: [
+        'res/feature/responsivepeak/' + version + '/res/skins/',
+        'res/feature/responsivebase/' + version + '/res/skins/',
+        'res/feature/responsivebase/common/res/skins/'
+      ],
+      doClear: false
+    };
+
+    if (server.useResponsiveConfigsFromServer()) {
+      var respOptions = getResponsiveOptions();
+      respOptions.getOptions(opts, function (config) {
+        setOptionsFromConfig(config, opts);
+        cb(opts);
+      });
+    } else {
+      cb(opts);
+    }
+  }
+
   return {
     process: function (patterns, dest, files) {
       return gulp.src(patterns)
-       .pipe(gutil.env.filterFiles(files))
-       .pipe(gulp.dest(dest));
+        .pipe(gutil.env.filterFiles(files))
+        .pipe(gulp.dest(dest));
     },
     compile: function (lr) {
-      var server = getServer();
-      var version = server.localSkinCompileVersion();
-      var feature = server.localSkinCompileFeature();
-      var skin = server.localSkinCompileSkin();
-      var dest = server.localSkinCompileDest();
-      var skinPath = 'res/feature/' + feature + '/' + version + '/res/skins/' + skin + '/sass/skin.scss';
-      var includePaths = [
-        'res/feature/responsivepeak/' + version + '/res/skins/',
-        'res/feature/responsivebase/' + version + '/res/skins/',
-        'res/feature/responsivebase/common/res/skins/'
-      ];
-
-      return gulp.src(skinPath)
-        .pipe(lr ? plumber()() : gutil.noop())
-        .pipe(sourcemaps().init())
-        .pipe(sass()({
-          includePaths: includePaths
-        }))
-        .pipe(sourcemaps().write())
-        .pipe(rename()(skin + '.css'))
-        .pipe(gulp.dest(dest))
-        .pipe(lr ? lr() : gutil.noop());
+      withOptions(function (opts) {
+        opts.lr = lr;
+        if (server.useResponsiveConfigsFromServer()) {
```

Comment:
@adamayres I tried to leave the logic you had the same in this script by adding this server.conf.json property check -- if you don't have the useResponsiveConfigsFromServer property set to true, it should behave like it did before. Let me know if you spot anything that doesn't look right or run into problems compiling.